common-lib/src/main/scala/com/gu/mediaservice/lib/auth/PermissionsHandler.scala

```diff
@@ -1,25 +1,31 @@
 package com.gu.mediaservice.lib.auth
 
+import com.gu.editorial.permissions.client._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext.Implicits.global
 
 
 object PermissionDeniedError extends Throwable("Permission denied")
 
-trait PermissionsHandler {
+object PermissionsHandler {
   val failFuture = Future.failed(PermissionDeniedError)
-  val permissionStore: PermissionStore
 
-  def validateUserWithPermissions(user: Principal, permission: PermissionType.PermissionType)
+  def validateUserWithPermissions(user: Principal, permission: Permission)
                                  (implicit ec: ExecutionContext): Future[Principal] =
-    getPermissionValForUser(permission, user) flatMap {
+    hasPermission(user, permission) flatMap {
       case true  => Future.successful(user)
       case false => failFuture
     }
 
-  def getPermissionValForUser(permission: PermissionType.PermissionType, user: Principal): Future[Boolean] = {
+  def hasPermission(user: Principal, permission: Permission) : Future[Boolean] = {
     user match {
       case u: PandaUser => {
-        permissionStore.hasPermission(permission, u.email)
+        Permissions.get(permission)(PermissionsUser(u.email)).map {
+          case PermissionGranted => true
+          case PermissionDenied => false
+
+        // fail open
+        } recover { case  _ => true }
```

Comment:
Just to check - if the call fail then we'll grant access? Assumption is we trust the user?
We should log the recover?